presidio-analyzer/presidio_analyzer/analyzer_engine.py

```diff
@@ -223,6 +227,116 @@ def analyze(
 
         return results
 
+    def analyze_iterator(
+        self,
+        list_of_texts: Iterable[Union[str, bool, float, int]],
+        language: str,
+        **kwargs,
+    ) -> List[List[RecognizerResult]]:
+        """
+        Analyze an iterable of strings.
+
+        :param list_of_texts: An list containing strings to be analyzed.
+        :param language: Input language
+        :param kwargs: Additional parameters for the `AnalyzerEngine.analyze` method.
+        """
+
+        # validate type
+        list_of_texts = (text for text in list_of_texts if self._validate_type(text))
+
+        # Process the texts as batch for improved performance
+        nlp_artifacts_batch: Iterator[
+            Tuple[str, NlpArtifacts]
+        ] = self.nlp_engine.process_batch(texts=list_of_texts, language=language)
+
+        list_results = []
+        for text, nlp_artifacts in nlp_artifacts_batch:
+            results = self.analyze(
+                text=str(text), nlp_artifacts=nlp_artifacts, language=language, **kwargs
+            )
+
+            list_results.append(results)
+
+        return list_results
+
+    @staticmethod
+    def _validate_type(obj: Any) -> bool:
+        if not type(obj) in (int, float, bool, str):
+            raise ValueError(
+                "Analyzer.analyze_iterator only works on primitive types "
+                "(int, float, bool, str). "
+                "Lists of objects are not yet supported."
+            )
+        return True
+
+    def analyze_dict(
```

Comment:
seems like a helper method or a special-case-wrapper for the analyze-iterator. more than a "core" analyzer functionality to me, but let's discuss at the review session.

Reply:
This is something the user is likely to use for any JSON file, or any tabular data that's turned into a dictionary of key:List[str]
See some examples here: https://github.com/microsoft/presidio/blob/2c68459bdd9f85c09dfe82277b006d14da42e0ff/docs/samples/python/batch_processing.ipynb

Documentation is still missing, wanted to get everybody's feedback before finalizing it.